events/class_tests/create_meeting_form.py

```diff
@@ -0,0 +1,195 @@
+        assert not view.meeting_participants_formset.is_valid()
+        assert expected_error in view.meeting_participants_formset.non_form_errors()
+
+    @pytest.mark.parametrize("invalid_optional_meeting_data,expected_error,is_formset_error", [
+        # blank end date
+        (
+                {'optional_meetings-TOTAL_FORMS': '1', 'optional_meetings-INITIAL_FORMS': '0',
+                 'optional_meetings-0-date_time_start': OPTIONAL_MEETING_DATE_START,
+                 'optional_meetings-0-date_time_end': '',
+                 },
+                "'Ending date cannot be blank'", False
+        ),
+        (
+                # blank start date
+                {'optional_meetings-TOTAL_FORMS': '1', 'optional_meetings-INITIAL_FORMS': '0',
+                 'optional_meetings-0-date_time_start': '',
+                 'optional_meetings-0-date_time_end': OPTIONAL_MEETING_DATE_END,
+                 },
+                "Starting date cannot be blank", False
+        ),
+        (
+                # dates not unique
+                {'optional_meetings-TOTAL_FORMS': '2', 'optional_meetings-INITIAL_FORMS': '0',
+                 'optional_meetings-0-date_time_start': OPTIONAL_MEETING_DATE_START,
+                 'optional_meetings-0-date_time_end': OPTIONAL_MEETING_DATE_END,
+                 'optional_meetings-1-date_time_start': OPTIONAL_MEETING_DATE_START,
+                 'optional_meetings-1-date_time_end': OPTIONAL_MEETING_DATE_END,
+                 },
+                "The optional meeting dates should be different", True
+        ),
+        (
+                # dates in the past
+                {'optional_meetings-TOTAL_FORMS': '1', 'optional_meetings-INITIAL_FORMS': '0',
+                 'optional_meetings-0-date_time_start': OPTIONAL_MEETING_DATE_START_IN_THE_PAST,
+                 'optional_meetings-0-date_time_end': OPTIONAL_MEETING_DATE_END_IN_THE_PAST,
+                 },
+                "Optional meeting dates cannot be in the past", True
+        )
+    ], ids=[
+        "blank end date",
+        "blank start date",
+        "dates not unique",
+        "dates in the past",
+    ]
+                             )
+    def test_invalid_optional_meeting_formset(self, invalid_optional_meeting_data, expected_error, is_formset_error):
+        view = self.create_meeting_view(optional_meeting_data=invalid_optional_meeting_data)
+        assert not view.optional_meetings_formset.is_valid()
+        if is_formset_error:
+            assert expected_error in view.optional_meetings_formset.non_form_errors()
+        else:
+            assert expected_error in f'{view.optional_meetings_formset.errors}'
+
+    def test_render_meeting_template(self, client, sign_in):
+        response = client.get(MEETING_CREATION_URL)
+        assert response.status_code == 200
+        assertTemplateUsed(response, MEETING_CREATION_HTML_PATH)
+
+    def create_meeting_view(self, user_id=None, event_data=None, optional_meeting_data=None,
```

Comment:
Can you make this a static function, as this function doesn't uses self

Reply:
Done